advisor/src/test/kotlin/advisors/OssIndexTest.kt

```diff
@@ -0,0 +1,158 @@
+                                            "7ea56ad4-8a8b-4e51-8ed9-5aad83d8efb1?component-type=maven" +
+                                            "&component-name=junit.junit&utm_source=mozilla&utm_medium=integration" +
+                                            "&utm_content=5.0"
+                                    ),
+                                    scoringSystem = "CVSS:3.0",
+                                    severity = "5.5"
+                                ),
+                                VulnerabilityReference(
+                                    url = URI("https://nvd.nist.gov/vuln/detail/CVE-2020-15250"),
+                                    scoringSystem = "CVSS:3.0",
+                                    severity = "5.5"
+                                )
+                            )
+                        )
+                    )
+                }
+            }
+        }
+
+        "handle a failure response from the server" {
+            WireMock.stubFor(
+                WireMock.post(WireMock.urlPathEqualTo(COMPONENTS_REQUEST_URL))
+                    .willReturn(
+                        WireMock.aResponse().withStatus(500)
+                    )
+            )
+            val ossIndex = OssIndex(ADVISOR_NAME, "http://localhost:${server.port()}")
+            val packages = COMPONENTS_REQUEST_IDS.map { Package.EMPTY.copy(id = it, purl = it.toPurl()) }
+
+            val result = ossIndex.retrievePackageFindings(packages).mapKeys { it.key.id }
+
+            result shouldNotBeNull {
+                keys should io.kotest.matchers.collections.containExactly(COMPONENTS_REQUEST_IDS)
+
+                COMPONENTS_REQUEST_IDS.forEach { pkg ->
+                    val pkgResults = getValue(pkg)
+                    pkgResults shouldHaveSize 1
+                    val pkgResult = pkgResults[0]
+                    pkgResult.vulnerabilities should io.kotest.matchers.collections.beEmpty()
+                    pkgResult.summary.issues shouldHaveSize 1
+                    val issue = pkgResult.summary.issues[0]
+                    issue.severity shouldBe Severity.ERROR
+                }
+            }
+        }
+    }
+})
+
+private const val ADVISOR_NAME = "OssIndexTest"
+
+private const val TEST_FILES_ROOT = "src/test/assets"
+private const val TEST_FILES_DIRECTORY = "oss-index"
+
+private val ID_HAMCREST = Identifier("Maven:org.hamcrest:hamcrest-core:1.3")
+private val ID_JUNIT = Identifier("Maven:junit:junit:4.12")
+
+private const val COMPONENTS_REQUEST_URL = "/api/v3/component-report"
+private val COMPONENTS_REQUEST_IDS = listOf(ID_HAMCREST, ID_JUNIT)
+
+private val COMPONENTS_REQUEST_JSON =
+    COMPONENTS_REQUEST_IDS.joinToString(prefix = "{ \"coordinates\": [", postfix = "] }") { "\"${it.toPurl()}\"" }
+
+private fun WireMockServer.stubComponentsRequest(responseFile: String) {
+    stubFor(
+        WireMock.post(WireMock.urlPathEqualTo(COMPONENTS_REQUEST_URL))
```

Comment:
Any reason to not import the static members of `WireMock` to make the expressions shorter?

Reply:
No specific reason, I just didn't care 😉 I've done it now, also did it consistently in the `VulnerableCodeTest` "sibling" test.